src/compiler/c.rs

```diff
@@ -1121,14 +1164,25 @@ impl<I: CCompilerImpl> Compilation for CCompilation<I> {
             ref env_vars,
             ..
         } = *self;
-        compiler.generate_compile_commands(
+
+        let (command, dist_command, cacheable) = compiler.generate_compile_commands(
             path_transformer,
             executable,
             parsed_args,
             cwd,
             env_vars,
             rewrite_includes_only,
-        )
+        )?;
+
+        let force_no_cache = env_vars
```

Comment:
same here

Reply:
Unfortunately this is also related to this feature, and I left an explanation in [this comment](https://github.com/trxcllnt/sccache/blob/282ea920deb5c2baf4b236804a2e359518c04004/src/compiler/nvcc.rs#L690-L702).

This behavior ensures we don't cache the same object file twice under two different hashes.

In the case where a call like `nvcc -c x.cu -o x.cu.o` is cached, `x.cu.o` is stored at a hash computed from the inputs. 

In the case that it's not cached, we generate a list of commands to produce `x.cu.o`, where the final command is the host compiler call that combines the fatbin + host code, i.e.:
```
<all other cicc, ptxas, fatbinary etc. calls>
gcc "$tmp/x.cudafe1.cpp" -o "x.cu.o"
```

We do want to run this host compiler call through sccache, so we can take advantage of distributed compilation, but doing that would normally cache `x.cu.o` with a hash computed from the gcc compilation flow. 

Since this is the object we're going to cache under the original hash computed for the outer nvcc call, and these objects can be large since they include the device code for all archs, we should avoid caching it twice.